src/algoliasearch.helper.js

```diff
@@ -247,6 +247,27 @@ AlgoliaSearchHelper.prototype.setState = function( newState ) {
 };
 
 /**
+ * Override the current state without triggering a change event.
+ * Do not use this method unless you know what you are doing. (see the example
+ * for a legit use case)
+ * @param {SearchParameters} newState the whole new state
+ * @return {AlgoliaSearchHelper}
+ * @example
+ *  helper.on( "change", function( state ){
+ *    // In this function you might want to find a way to store the state in the url/history
+ *    updateYourURL( state );
+ *  } );
+ *  window.onpopstate = function( event ){
+ *    // This is naive though as you should check if the state is really defined etc.
+ *    helper.overrideState( event.state ).search();
+ *  }
+ */
+AlgoliaSearchHelper.prototype.overrideState = function( newState ) {
```

Comment:
We should name it with something less ambiguous. It is not clear from the method signature what the difference between `setState` and `overrideState` will be.

I haven't found a better name yet, but here are some ideas:
- `setStateWithoutTriggeringChangeEvent`
- `setStateSilent`
- `setState(newState, { 'silent': true })`
- `_setState`